### ISSUE TITLE
AnnotationSpec throwing java.lang.reflect.InvocationTargetException instead of User Exception

### ISSUE THREAD

**Issue by Kerooker** [kotlintest/kotlintest#539]
When executing a test with AnnotationSpec:

```

class Foo : AnnotationSpec() {
  class FooException: Exception()


  @Test
  fun foo() {
    throw FooException()
  }
}
```

I expected the stacktrace to contain my exception and my line at the top, but instead it returns a weird Stacktrace.

This happened after changing it's execution to `invokeSuspend`

```
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:71)
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Instance.call(CallerImpl.kt:87)
	at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:106)
	at io.kotlintest.specs.AbstractAnnotationSpec$createTestCase$$inlined$let$lambda$1.invokeSuspend(AbstractAnnotationSpec.kt:76)
.....

```

**Comment by Kerooker** [comment]
I just discovered this isn't actually related to `invokeSuspend`, therefore it doesn't have anything to do with `3.2.0-RC2`. It's just a bug in `AnnotationSpec`

Should we point it to 3.2.0 anyways, @sksamuel ?

**Comment by sksamuel** [comment]
Yeah can put it on 3.2.0 unless someone asks for it on 3.1

**Comment by Kerooker** [comment]
I meant if this should go for 3.2 or 3.3, as it's unrelated to 3.2

**Comment by sksamuel** [comment]
Yes it's a bug and we've not yet released 3.2.0 so definitely target that.
